src/graphql/middlewares/auth.js

```diff
@@ -1,22 +1,27 @@
-const config = require('config');
-const logger = require('../lib/logger');
+const redisClient = require('../lib/redis');
 
-let apikeyusers = config.get('apikeyusers');
-try {
-  apikeyusers = JSON.parse(apikeyusers);
-} catch (err) {
-  logger.error(`Cannot parse "${apikeyusers}" in json format`);
-  logger.error(err);
-}
 /**
  * check the user's api key
  * @param {Object} req - HTTP request
  * @param {Object} res - HTTP response
  * @param {function} next - do the following
  * @returns {Object|function} res or next
  */
-const checkAuth = (req, res, next) => {
-  if (!apikeyusers.includes(req.get('X-API-KEY'))) {
+const checkAuth = async (req, res, next) => {
+  // TODO check in query
+  const apikey = req.get('X-API-KEY');
+
+  if (!apikey) {
+    return res.status(401).json({ message: 'Not authorized' });
+  }
+
+  const key = await redisClient.get(apikey);
+  const keyParsed = JSON.parse(key);
```

Comment:
Don't forget to handle errors, Express does not support promises. And keep in mind that `JSON.parse(key)` will throw if `key` does not exist.